R/deduplicate.R

```diff
@@ -0,0 +1,206 @@
+##' Deduplicate an orderly archive.  Deduplicating an orderly archive
+##' will replace all files that have the same content with "hard
+##' links".  This is reliable only on Unix-like systems.  If you use
+##' this feature, it is \emph{very important} that you treat your
+##' orderly archive as read-only (though you should be anyway) as
+##' changing one copy of a linked file changes all the other instances
+##' of it - the files are literally the same file.
+##'
+##' This function will alter your orderly archive.  Ordinarily this is
+##' not something that should be done, so we try to be careful.  In
+##' order for this to work, it is \emph{very important} to treat your
+##' orderly archive as read-only generally.  If your canonical orderly
+##' archive is behind OrderlyWeb this will almost certainly be the
+##' case already.
+##'
+##' With "hard linking", two files with the same content can be
+##' updated so that both files point at the same physical bit of data
+##' (see \href{https://en.wikipedia.org/wiki/Hard_link}{this wikipedia
+##' page for more information}).  This is great, as if the file is
+##' large, then only one copy needs to be stored.  However, this means
+##' that if a change is made to one copy of the file, it is
+##' immediately reflected in the other, but there is nothing to
+##' indicate that the files are linked!
+##'
+##' This approach is worth exploring if you have large files that are
+##' outputs of one report and inputs to another, or large inputs
+##' repeatedly used in different reports, or outputs that end up being
+##' the same in multiple reports.  If you run the deduplication with
+##' \code{dry_run = TRUE}, an indication of the savings will be
+##' printed.
+##'
```

Comment:
General OS comment - For the sake of getting orderly on to CRAN, how important is it that orderly should be possible to run on Windows? It sounds like this would be a significant (and awkward) programming investment, and probably one for me to tackle if that were a desired thing.

Reply:
There is probably nothing stopping this running on windows, and if you wanted to check it please do.  The issue can you see how this gets on with deduplicating on a win7/win10 machine by removing the check. The things to watch out for are:

* how does the inode check go?
* does the hardlink work?
* can we tell that the hardlink worked?